docker-compose.yml

```diff
@@ -0,0 +1,73 @@
+services:
+    laravel.test:
+        build:
+            context: ./vendor/laravel/sail/runtimes/8.2
+            dockerfile: Dockerfile
+            args:
+                WWWGROUP: '${WWWGROUP}'
+        image: sail-8.2/app
+        extra_hosts:
+            - 'host.docker.internal:host-gateway'
+        ports:
+            - '${APP_PORT:-80}:80'
+            - '${VITE_PORT:-5173}:${VITE_PORT:-5173}'
+        environment:
+            WWWUSER: '${WWWUSER}'
+            LARAVEL_SAIL: 1
+            XDEBUG_MODE: '${SAIL_XDEBUG_MODE:-off}'
+            XDEBUG_CONFIG: '${SAIL_XDEBUG_CONFIG:-client_host=host.docker.internal}'
+            IGNITION_LOCAL_SITES_PATH: '${PWD}'
+        volumes:
+            - '.:/var/www/html'
+        networks:
+            - sail
+        depends_on:
+            - mysql
+            - redis
+    mysql:
+        image: 'mysql/mysql-server:8.0'
+        ports:
+            - '${FORWARD_DB_PORT:-3306}:3306'
+        environment:
+            MYSQL_ROOT_PASSWORD: '${DB_PASSWORD}'
+            MYSQL_ROOT_HOST: '%'
+            MYSQL_DATABASE: '${DB_DATABASE}'
+            MYSQL_USER: '${DB_USERNAME}'
+            MYSQL_PASSWORD: '${DB_PASSWORD}'
+            MYSQL_ALLOW_EMPTY_PASSWORD: 1
+        volumes:
+            - 'sail-mysql:/var/lib/mysql'
+            - './vendor/laravel/sail/database/mysql/create-testing-database.sh:/docker-entrypoint-initdb.d/10-create-testing-database.sh'
+        networks:
+            - sail
+        healthcheck:
+            test:
+                - CMD
+                - mysqladmin
+                - ping
+                - '-p${DB_PASSWORD}'
+            retries: 3
+            timeout: 5s
+    redis:
+        image: 'redis:alpine'
+        ports:
+            - '${FORWARD_REDIS_PORT:-6379}:6379'
+        volumes:
+            - 'sail-redis:/data'
+        networks:
+            - sail
+        healthcheck:
+            test:
+                - CMD
+                - redis-cli
+                - ping
+            retries: 3
+            timeout: 5s
```

Comment:
coba mas @rockavoldy bikin service baru untuk running script migrate and seednya

```
  laravel.test::
     image: sail-app
     ....
     ....
  
   migrate-seed:
        image: sail-app
        depends_on:
            - mysql
        command: >
            bash -c "
                while ! mysqladmin ping -hmysql --silent; do
                    sleep 1
                done
                ./vendor/bin/sail artisan migrate &&
                ./vendor/bin/sail artisan db:seed
            "
        volumes:
            - .:/var/www/html
```

Reply:
Setelah coba-coba buat service baru di docker-composenya, cara ini gak possible mas @muhfaris , karena sail ini harus dijalankan di host-nya, jadi sail ini lebih seperti wrapper untuk `docker compose`

sudah coba juga untuk ubah command dari pertama run si container `laravel.test`, dan gak berhasil juga

akhirnya di check si mysql servicenya sudah bisa nerima connection atau belum dengan `sail exec` ke container mysql-nya, sepertinya lebih reliable karena ternyata sail ini wrappernya docker compose, jadi baca container name nya yang ada di file `docker-compose.yml`, bukan seperti `docker exec` yang pakai service name-nya